packages/engine/src/datastore/table/state/mod.rs

```diff
@@ -38,23 +38,35 @@ pub struct Inner {
     local_meta: Meta,
 
     num_agents: usize,
+
+    sim_config: Arc<SimRunConfig>,
 }
 
-impl Inner {
-    /// Creates a new State object from a collection of groups of `AgentState`.
```

Comment:
Actually I think having a separate function for conversion from groups to batches was still useful for readability. It could just be on `State` as `from_agent_groups` instead of on `Inner`. Inlining the function makes it rather long.